# Array indices from 0 to max match joints from base joint to last joint
int8[] modules